File: app/src/main/java/com/lazycouple/restapiclient/data/DataSource.java
package com.lazycouple.restapiclient.data;

import com.lazycouple.restapiclient.db.model.Api;
import com.lazycouple.restapiclient.db.model.Parameter;

import java.util.List;

import io.realm.Realm;
import rx.Observable;

/**
 * Created by amuyu on 2017. 6. 13..
 */

public interface DataSource {


    interface LoadHistoriesCallback {
        void onHistoriesCallback(List<String> histories);
    }

    Observable<List<String>> getHistories();
    Observable<List<Api>> getApiHistories(Realm realm);
    void addApi(String url, List<Parameter> parameters);
    Observable<Api> getApi(Realm realm, String id);
    void clearApiHistories();


}

File: app/src/main/kotlin/com/lazycouple/restapiclient/data/DataSource.kt
package com.lazycouple.restapiclient.data

import com.lazycouple.restapiclient.db.model.Api
import com.lazycouple.restapiclient.db.model.Parameter
import com.lazycouple.restapiclient.ui.presenter.RestRequestPresenter

import io.realm.Realm
import rx.Observable

/**
 * Created by amuyu on 2017. 6. 13..
 */

interface DataSource {


    interface LoadHistoriesCallback {
        fun onHistoriesCallback(histories: List<String>)
    }

    fun getHistories(): Observable<List<String>>?
    fun getApiHistories(realm: Realm): Observable<List<Api>>
    fun addApi(url: String, method: RestRequestPresenter.Method, parameters: List<Parameter>)
    fun getApi(realm: Realm, id: String): Observable<Api>
    fun clearApiHistories()


}
